src/OpenColorIO/CPUInfoConfig.h.in

```diff
@@ -2,14 +2,50 @@
 // Copyright Contributors to the OpenColorIO Project.
 
 
-#cmakedefine OCIO_ARCH_X86
-
-#cmakedefine01 OCIO_USE_SSE2
-#cmakedefine01 OCIO_USE_SSE3
-#cmakedefine01 OCIO_USE_SSSE3
-#cmakedefine01 OCIO_USE_SSE4
-#cmakedefine01 OCIO_USE_SSE42
-#cmakedefine01 OCIO_USE_AVX
-#cmakedefine01 OCIO_USE_AVX2
-#cmakedefine01 OCIO_USE_AVX512
-#cmakedefine01 OCIO_USE_F16C
+#cmakedefine01 OCIO_ARCH_X86
+
+// Relevant only for arm64 architecture.
+#if defined(__aarch64__)
+    #cmakedefine01 OCIO_USE_SSE2NEON
+#endif
+
+// On the Apple platform, a universal build is created for both x86_64 and arm64 architectures. 
+// CMake will run only one "configure" step, build for x86_64 and arm64, and then create a 
+// single binary that includes both architectures.
+
+// This means that for a universal build, the OCIO_USE_SSEx, OCIO_USE_AVXx, OCIO_USE_F16C, etc. 
+// flags can be enabled simultaneously. Therefore, we need to check whether we are currently 
+// building the x86_64 side or the arm64 side. This can be done by checking the OCIO_ARCH_X86 and 
+// aarch64 defines.
+
+// Building for x86_64 processor on a non-ARM host architecture
+// OR Building on/for an ARM architecture and using SSE2NEON.
+#if (OCIO_ARCH_X86 && !defined(__aarch64__)) || (defined(__aarch64__) && OCIO_USE_SSE2NEON)
+    #cmakedefine01 OCIO_USE_SSE2
+    #cmakedefine01 OCIO_USE_SSE3
+    #cmakedefine01 OCIO_USE_SSSE3
+    #cmakedefine01 OCIO_USE_SSE4
+    #cmakedefine01 OCIO_USE_SSE42
```

Comment:
Building non universal arm64, I still needed to specify `-DOCIO_USE_SSE2=ON` etc to turn all these on. `-DOCIO_USE_SSE2NEON=ON` should probably imply all these SSE features are ON.

Reply:
They are not turning on for me when I build natively on arm64. I've try cleaning my env several times.
I'm just doing the basic `mkdir build && cd build && cmake .. && cmake --build .`

and my `CPUInfoConfig.h` shows

```
#if (OCIO_ARCH_X86 && !defined(__aarch64__)) || (defined(__aarch64__) && OCIO_USE_SSE2NEON)
    #define OCIO_USE_SSE2 0
    #define OCIO_USE_SSE3 0
    #define OCIO_USE_SSSE3 0
    #define OCIO_USE_SSE4 0
    #define OCIO_USE_SSE42 0
```

It looks like its because `CMAKE_OSX_ARCHITECTURES` isn't set to anything when I build it this way.
That section in [CMakeLists.txt](https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1828/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR192-R202) need it to be `arm64`.

`(APPLE AND "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm64" AND "${CMAKE_OSX_ARCHITECTURES}" MATCHES "arm64")`

`cmake ..  -DCMAKE_OSX_ARCHITECTURES=arm64` works.

Should we set CMAKE_OSX_ARCHITECTURES = CMAKE_SYSTEM_PROCESSOR if undefined?

Reply:
Oh right, I get it now. There's a missing condition in the CMakeLists.txt. I'll update it.

Reply:
The latest commit should have fixed the issue (I've tested it on my M1). @markreidvfx

Reply:
That fixes it for me on my M2 mac too.